db/products.js

```diff
@@ -0,0 +1,82 @@
+const productCollection = {
+  products: [],
+};
+let productId = 1;
+
+function create(obj) {
+  const hasKeys = obj.name && obj.price && obj.inventory;
+
+  if (hasKeys) {
+    obj.id = productId;
+    productCollection.products.push(obj);
+    productId++;
+    return `{ "success": true}`;
+  } else if (!hasKeys) {
+    return `{ "success": false}`;
+  }
+}
+
+function retrieveAll() {
+  return productCollection;
+}
+
+function retrieveOne(productId) {
+  // check that productID is in collection.
+  const productIndex = productCollection.products.findIndex((elem) => elem.id === Number(productId));
+  if (productIndex === -1) {
+    return `{ "success": false. Product ID not found }`;
```

Comment:
Generally w/ this pattern we'd prefer the status and the message to be separate for easier parsing on the front end.
```
    return { "success": false, "message": "Product ID not found "};

```